plasmapy/diagnostics/tests/test_charged_particle_radiography.py

```diff
@@ -459,26 +458,203 @@ def test_run_options():
+                (tracker_obj_simulated,),
+                {},
+                {
+                    "xrange": [-0.036934532101889815, 0.03702186098974771] * u.m,
+                    "yrange": [-0.03697401811598356, 0.037007901161403144] * u.m,
+                    "bins": (200, 200),
+                },
+            ),
+            (
+                # From a dict
+                (sim_results,),
+                {},
+                {
+                    "xrange": [-0.036934532101889815, 0.03702186098974771] * u.m,
+                    "yrange": [-0.03697401811598356, 0.037007901161403144] * u.m,
+                    "bins": (200, 200),
+                },
+            ),
+            (
+                # From a dict
+                (sim_results,),
+                {"size": np.array([[-1, 1], [-1, 1]]) * 30 * u.cm, "bins": (200, 60)},
+                {
+                    "xrange": [-0.3, 0.3] * u.m,
+                    "yrange": [-0.3, 0.3] * u.m,
+                    "bins": (200, 60),
+                },
+            ),
+        ],
+    )
+    def test_intensity_histogram(self, args, kwargs, expected):
+        """Test several valid use cases."""
+        results = cpr.synthetic_radiograph(*args, **kwargs)
+
+        assert len(results) == 3
+
+        x = results[0]
+        assert isinstance(x, u.Quantity)
+        assert x.unit == u.m
+        assert x.shape == (expected["bins"][0],)
+        assert np.isclose(np.min(x), expected["xrange"][0], rtol=1e4)
+        assert np.isclose(np.max(x), expected["xrange"][1], rtol=1e4)
+
+        y = results[1]
+        assert isinstance(y, u.Quantity)
+        assert y.unit == u.m
+        assert y.shape == (expected["bins"][1],)
+        assert np.isclose(np.min(y), expected["yrange"][0], rtol=1e4)
+        assert np.isclose(np.max(y), expected["yrange"][1], rtol=1e4)
+
+        histogram = results[2]
+        assert isinstance(histogram, np.ndarray)
+        assert histogram.shape == expected["bins"]
+
+    def test_optical_density_histogram(self):
+        """
+        Test the optical density calculation is correct and stuffed
+        with numpy.inf when the intensity is zero.
+        """
+        bins = (200, 60)
+        size = np.array([[-1, 1], [-1, 1]]) * 30 * u.cm
+
+        intensity_results = cpr.synthetic_radiograph(
+            self.sim_results, size=size, bins=bins
+        )
+        od_results = cpr.synthetic_radiograph(
+            self.sim_results, size=size, bins=bins, optical_density=True
+        )
+
+        assert np.allclose(intensity_results[0], od_results[0])
+        assert np.allclose(intensity_results[1], od_results[1])
+
+        intensity = intensity_results[2]
+        zero_mask = intensity == 0
+        i0 = np.mean(intensity[~zero_mask])
+        od = -np.log10(intensity / i0)
+
+        assert np.allclose(od[~zero_mask], od_results[2][~zero_mask])
+        assert np.all(np.isposinf(od_results[2][zero_mask]))
+
+
+def test_saving_output(tmp_path):
```

Comment:
```suggestion
def test_saving_output(tmp_path):
    """Test behavior of Tracker.save_results."""
```

Test functions and classes still need docstrings.  It's just that docstrings are written for the programmer/contributor and not the user.

Reply:
I've no issue adding this, but I don't think this docstring adds any information that isn't already in the name of the function.

Reply:
It does because the function name does not indicate that you're testing the `Tracker.save_results` method.  This test file is testing several different radiography functions, methods, and classes, so one can not just assume this test function is working on the `Tracker` class.